D/D.sublime-syntax

```diff
@@ -8,339 +8,2031 @@ file_extensions:
+        - match: \*/
+          scope: punctuation.definition.comment.d
           pop: true
-        - include: $top_level_main
-    - match: |-
-        (?x)^\s*
-        ((?:\b(private|package|protected|public|export|static|final|native|synchronized|abstract|const)\b\s*)*) # modifier
-        (struct)\s+
-        (\w+)\s* # identifier
-        (?:\(\s*([^\)]+)\s*\)|)\s* # Template type
+    - match: (///?).*$
+      scope: comment.line.double-slash.d
       captures:
-        1: storage.modifier.d
-        3: storage.type.d
-        4: entity.name.type.struct.d
-        5: storage.type.template.d
+        1: punctuation.definition.comment.d
+    - include: nested-comment
+  nested-comment:
+    - match: /\+\+?
+      scope: punctuation.definition.comment.d
       push:
-        - meta_scope: meta.definition.struct.d
-        - match: "(?={|//|;)"
+        - meta_scope: comment.block.nested.d
+        - match: \+/
+          scope: punctuation.definition.comment.d
           pop: true
-        - match: \b(_|:)\b
-          captures:
-            1: storage.modifier.d
-          push:
-            - meta_scope: meta.definition.class.extends.d
-            - match: "(?={)"
-              captures:
-                1: storage.modifier.d
-              pop: true
-            - include: all-types
-    - match: |-
-        (?x)^\s*
-        ((?:\b(private|package|protected|public|export|static|final|native|synchronized|abstract|threadsafe|transient|shared|__gshared)\b\s*)*) # modifier
-        (\b(this))\s* # identifier
-        (?!.*;)  # abort if line has a ;
-        (?=\()
+        - include: nested-comment
+
+  number-in:
+    - match: '{{number_lookahead}}'
+      push: number
+  number-opt:
+    - include: floating-point-opt
+    - include: integer-opt
+  number:
+    - include: number-opt
+    - include: not-whitespace-illegal
+
+  integer:
+    - include: integer-opt
+    - include: not-whitespace-illegal
+  integer-opt:
+    - match: '\b(0|[1-9][0-9_]*)({{integer_suffix}})?\b'
+      captures:
+        1: constant.numeric.integer.d
+        2: storage.type.integer.d
+      pop: true
+    - match: '\b(0[bB])([01][01_]*)({{integer_suffix}})?\b'
       captures:
         1: storage.modifier.d
-        3: entity.name.function.constructor.d
-      push:
-        - meta_scope: meta.function.d meta.function.constructor.d
-        - include: function_assertions
-        - include: $top_level_main
-    - match: |-
-        (?x)
-                    (?:  ^                                 # begin-of-line
-                      |  (?: (?<!else|new|=) )             #  or word + space before name
-                    )
-                  ((?:\b(?:private|package|protected|public|export|static|final|native|synchronized|abstract|threadsafe|transient|override)\b\s*)*) # modifier
-                    (~this) # actual name
-                     \s*(\()                           # start bracket or end-of-line
+        2: constant.numeric.binary.d
```

Comment:
I can understand this kind of thinking from perspective of compatibility with existing color schemes, but it lead to weird scope names in the past and mixes up the approach to define well structured standardized scope names with creating color schemes. How such scopes are colorized depend very much on the color scheme being actually used. It might look well with your prefered color scheme, but might look totally odd with another one. Hence adapting the scope names to existing color schemes is the wrong way round.

First step is to define clearly structured scope names, which hopefully - with much effort left for future changes - reflect comparable structures of all the different syntaxes the same way and then create good color schemes upon that standard, which make look all syntaxes the same way.

The new `sublime-color-scheme` format makes tweaking existing color schemes very easy anyway.

Reply:
Definitely agree (hence the change), was just making light of my thought process at the time of writing